core/src/test/java/io/substrait/type/proto/ExtensionRoundtripTest.java

```diff
@@ -232,4 +227,68 @@ void cross() {
             .build();
     verifyRoundTrip(rel);
   }
+
+  @Nested
+  class ExtensionThroughExpression {
+    // There are some expression that can contains relations.
+    // Check that custom extensions in these relations can be handled.
+
+    Rel baseTable =
+        b.namedScan(
+            Stream.of("test_table").collect(Collectors.toList()),
+            Stream.of("test_column").collect(Collectors.toList()),
+            Stream.of(TypeCreator.REQUIRED.I64).collect(Collectors.toList()));
```

Comment:
Minor, but I'm curious why this bothers creating a stream and collecting to list rather than

```
List.of("test_table"),
List.of("test_column"),
...
```

or is substrait-java targeting an older jdk version which doesn't have `List.of()`?

Reply:
Java 8 also has support for `Collections.singletonList("test_table")`

Reply:
> is substrait-java targeting an older jdk version which doesn't have List.of()?

Exactly that 
> Java 8 also has support for Collections.singletonList("test_table")

TIL. I might use that in the future.